Mathlib/CategoryTheory/Sites/Preserves.lean

```diff
@@ -0,0 +1,256 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.CategoryTheory.Limits.Opposites
+import Mathlib.CategoryTheory.Limits.Preserves.Shapes.Products
+import Mathlib.CategoryTheory.Limits.Shapes.DisjointCoproduct
+import Mathlib.CategoryTheory.Sites.SheafOfTypes
+import Mathlib.Tactic.ApplyFun
+
+universe v u w
+
+open CategoryTheory Limits Opposite
+
+variable {C : Type u} [Category.{v} C] (F : Cᵒᵖ ⥤ Type (max u v)) [HasInitial C]
+    (hF : (Presieve.ofArrows (X := ⊥_ C) Empty.elim instIsEmptyEmpty.elim).IsSheafFor F)
+
+instance : (Presieve.ofArrows (X := ⊥_ C) Empty.elim instIsEmptyEmpty.elim).hasPullbacks := by
+  constructor
+  intro _ _ _ hf
+  cases' hf with i
+  exact Empty.elim i
+
+instance : IsEmpty (Σ(Y : C), {f : Y ⟶ ⊥_ C //
+    (Presieve.ofArrows (X := ⊥_ C) Empty.elim instIsEmptyEmpty.elim) f}) := by
+  constructor
+  rintro ⟨_, _, ⟨i⟩⟩
+  exact Empty.elim i
+
+lemma isoTerminalComparisonOfIsSheafForEmpty : IsIso (terminalComparison F) := by
```

Comment:
Might be related: `Sheaf.isTerminalOfBotCover`

Reply:
Thanks for pointing this out! It's not exactly what I need, but I could imitate that proof to get a shorter one here